allennlp/service/predictors/coref.py

```diff
@@ -0,0 +1,61 @@
+from overrides import overrides
+
+import spacy
+
+from allennlp.common.util import JsonDict, sanitize
+from allennlp.data import DatasetReader, Instance
+from allennlp.models import Model
+from allennlp.service.predictors.predictor import Predictor
+
+
+@Predictor.register("coreference-resolution")
+class CorefPredictor(Predictor):
+    """
+    Wrapper for the :class:`~allennlp.models.coreference_resolution.CoreferenceResolver` model.
+    """
+    def __init__(self, model: Model, dataset_reader: DatasetReader) -> None:
+        super().__init__(model, dataset_reader)
+        self._spacy = spacy.load("en", ner=False, tagger=False, vectors=False)
+
+    def _json_to_instance(self, json: JsonDict) -> Instance:
+        # We're overriding `predict_json` directly, so we don't need this.  But I'd rather have a
```

Comment:
Can you just use this?  It looks like all `predict_json` is doing is converting the json to an instance, then doing what `super.predict_json` does.

Reply:
I want to return the tokenised document - I don't think we want to do this by reconstructing the document from the indexed version in the model, but we could change the signature of `_json_to_instance` to be `Instance, Optional[JsonDict]`, so that you could pass things you don't want to be modified through to the output. I'm happy with just using the predict_json, but happy to make that change if you prefer.

Reply:
Hmm, yeah, what you have is fine.  The fact that we need to do this, both here and for the SRL model, makes me wonder if we should change the API a bit.  But that's an issue for a separate PR.